kitura-init.js

```diff
@@ -119,32 +128,68 @@ function renameProject() {
-    if (renameClean.status !== 0) {
-        console.error(chalk.red('Error: ') + 'could not rename project.');
-        console.error(renameClean.stderr.toString());
-        process.exit(renameClean.status);
+    //Creates an artifact based on the current contents of the project files, which contains the generator placeholders.
+    const lowercaseCurrentArtifacts = getArtifactsFor(oldProjNameCleanLowercase, "lowercase");
+    //Creates an artifact that will be used to replace the lowercaseCurrentArtifacts with the new project name.
+    const lowercaseNewArtifacts = getArtifactsFor(projNameClean, "lowercase");
+
+    //Creates an artifact based on the current contents of the project files, which contains the generator placeholders.
+    const uppercaseCurrentArtifacts = getArtifactsFor(oldProjName, "uppercase");
+    //Creates an artifact that will be used to replace the uppercaseCurrentArtifacts with the new project name.
+    const uppercaseNewArtifacts = getArtifactsFor(projName, "uppercase");
+
+    for (const fileName of Object.keys(lowercaseCurrentArtifacts)) {
+      //We need to add a special case for chart as the name of the directory within chart depends on the name of the project.
+      if (fileName === "chart") {
+         const filePath = `chart/${projNameCleanLowercase}/values.yaml`
+         replaceProjectStringsInArtifact(filePath, lowercaseCurrentArtifacts[fileName], lowercaseNewArtifacts[fileName]);
+      } else {
+        replaceProjectStringsInArtifact(fileName, lowercaseCurrentArtifacts[fileName], lowercaseNewArtifacts[fileName]);
+      }
     }
-    var renameLowercase = spawnSync('find', ['.', '-exec', 'sed', '-i', '', 's/' + oldProjNameLowercase + '/' + projNameLowercase + '/g', '{}', ';'])
-    if (renameClean.status !== 0) {
-        console.error(chalk.red('Error: ') + 'could not rename project.');
-        console.error(renameClean.stderr.toString());
-        process.exit(renameClean.status);
+    for (const fileName of Object.keys(uppercaseCurrentArtifacts)) {
+      replaceProjectStringsInArtifact(fileName, uppercaseCurrentArtifacts[fileName], uppercaseNewArtifacts[fileName]);
     }
-    var renameLowercaseClean = spawnSync('find', ['.', '-exec', 'sed', '-i', '', 's/' + oldProjNameCleanLowercase + '/' + projNameCleanLowercase + '/g', '{}', ';'])
-    if (renameClean.status !== 0) {
-        console.error(chalk.red('Error: ') + 'could not rename project.');
-        console.error(renameClean.stderr.toString());
-        process.exit(renameClean.status);
+}
+
+function getArtifactsFor(name, valueCase) {
+  return getArtifacts(name)[valueCase];
+}
+
+/*
+  Objects containing key values pairs of the files mapped to all the lines that require a change in that file.
+  We have a mapping for both the lowercased generatorswiftserverprojects as well as the uppercased Generator-Swiftserver-Projects.
+*/
+function getArtifacts(name) {
```

Comment:
So, is this fundamentally a good idea? We have a hard-coded list of places we need to go through and change. What happens when the generator changes and there's another file that needs editing? We'll just break won't we?

Wouldn't it be better to run `replaceInFile` over each file, so we match everything?